lib/shoryuken/manager.rb

```diff
@@ -154,61 +128,41 @@ def real_thread(proxy_id, thr)
 
     private
 
-    def build_processor
-      processor = Processor.new_link(current_actor)
-      processor.proxy_id = processor.object_id
-      processor
-    end
-
-    def restart_queue!(queue)
-      return if stopped?
-
-      unless @queues.include? queue
-        logger.debug { "Restarting '#{queue}'" }
-
-        @queues << queue
-
-        if @fetcher_paused
-          logger.debug { 'Restarting fetcher' }
+    def assign(queue, sqs_msg)
+      watchdog('Manager#assign died') do
+        logger.debug { "Assigning #{sqs_msg.message_id}" }
 
-          @fetcher_paused = false
+        processor = @ready.pop
+        @busy << processor
 
-          dispatch
-        end
+        processor.async.process(queue, sqs_msg)
       end
     end
 
-    def current_queue_weight(queue)
-      queue_weight(@queues, queue)
+    def dispatch_batch(queue)
+      batch = fetcher.fetch(queue, BATCH_LIMIT)
```

Comment:
@mariokostelac I wouldn't introduce `BATCH_LIMIT`, because we don't have a such thing. In this case we want: give the maximum you can. Just to make it simpler for future maintenance.